src/features/Search/pages/VisualNovelSearchPage/hooks/useQueryParams.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  DelimitedArrayParam,
+  NumberParam,
+  StringParam,
+  useQueryParams,
+  withDefault,
+  QueryParamConfig,
+  SetQuery,
+} from 'use-query-params';
+import { VisualNovelSearchOptions } from '../../../../../api/services/visualNovelsService';
+import { Language } from '../../../../../models/language';
+import { Platform } from '../../../../../models/platform';
+
+/** Uses a comma to delimit entries. */
+const ReleasedRangeParam: QueryParamConfig<
+VisualNovelSearchOptions['releasedRange'],
+VisualNovelSearchOptions['releasedRange']
+> = {
+  encode(releasedRange) {
+    return JSON.stringify({
+      startDate: releasedRange?.startDate?.toISOString() ?? null,
+      endDate: releasedRange?.endDate?.toISOString() ?? null,
+    });
+  },
+  decode(str) {
+    if (str == null || Array.isArray(str)) {
+      return undefined;
+    }
+    try {
+      const range = JSON.parse(str);
+      return {
+        startDate: range?.startDate != null ? new Date(range.startDate) : undefined,
+        endDate: range?.endDate != null ? new Date(range.endDate) : undefined,
+      };
+    } catch (e: unknown) {
+      return undefined;
+    }
```

Comment:
Can we just make `startDate` and `endDate` as separate properties?

Reply:
It would be too much of a hassle 😩, I would like to keep `realeaseRange`, but try to implement more readable query string.